pyOCD/utility/conversion.py

```diff
@@ -64,18 +64,7 @@ def float2int(data):
 
 ## @brief create 8-digit hexadecimal string from 32-bit register value.
 def intToHex8(val):
-    val = hex(int(val))[2:]
-    size = len(val)
-    r = ''
-    for i in range(8-size):
-        r += '0'
-    r += str(val)
-
-    resp = ''
-    for i in range(4):
-        resp += r[8 - 2*i - 2: 8 - 2*i]
-
-    return resp
+    return "%08X" % int(val)
```

Comment:
Good catch.  Reverting the code makes the test case fail, so that's no good.  I'll revert the implementation or provide an implementation that passes this test case:

```

    def test_intToHex8(self):
        self.assertEqual(intToHex8(0x0102ABCD), "cdab0201")
```